### PR TITLE
Add managed group support to role principals

### PR DESCRIPTION
**Pull request by jefferai**

### CODE REVIEW

**Review by vancluever — APPROVED** [review]
Long, but pretty much groked! LGTM

File: internal/db/schema/migrations/postgres/9/04_oidc_managed_group_principal_role.up.sql [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+
+create trigger immutable_role_principal
+before update on iam_managed_group_role
+  for each row execute procedure iam_immutable_role_principal();
+
+create trigger default_create_time_column
+before insert on iam_managed_group_role
+  for each row execute procedure default_create_time();
+
+-- iam_principle_role provides a consolidated view all principal roles assigned
```
[the remark this answers is not in the document]
Comment (jefferai) [reply internal/db/schema/migrations/postgres/9/04_oidc_managed_group_principal_role.up.sql:28]: hahaha...thanks!

File: internal/intglobals/prefixes.go [hunk 2]
```diff
@@ -0,0 +1,12 @@
+package intglobals
+
+// This set of consts is intended to be a place to collect commonly-used
```
[the remark this answers is not in the document]
Comment (jefferai) [reply internal/intglobals/prefixes.go:3]: Done!

File: internal/iam/repository_principal_role_ext_test.go [hunk 3]
```diff
@@ -0,0 +1,251 @@
...
+		wantSetGrps = append(wantSetGrps, newGrp.PublicId)
+		wantSetManagedGrps = append(wantSetManagedGrps, newManagedGrp.PublicId)
+
+		got, err := repo.PrincipalsToSet(context.Background(), r, wantSetUsers, wantSetGrps, wantSetManagedGrps)
+		require.NoError(err)
+		assertSetResults(t, got, []string{newUser.PublicId}, []string{newGrp.PublicId}, []string{newManagedGrp.PublicId}, wantDeleteUsers, wantDeleteGrps, wantDeleteManagedGrps)
+	})
+}
+
+func assertSetResults(t *testing.T, got *iam.PrincipalSet, wantAddUsers, wantAddGroups, wantAddManagedGroups, wantDeleteUsers, wantDeleteGroups, wantDeleteManagedGroups []string) {
```
[the remark this answers is not in the document]
Comment (jefferai) [reply internal/iam/repository_principal_role_ext_test.go:201]: Oh, I didn't make this. I think you did. :-)  This is just moved to this new file from its old location :-D

**Pull request merged**